libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/NavigationRouteEx.kt

```diff
@@ -0,0 +1,163 @@
+@file:JvmName("NavigationRouteUtils")
+
+package com.mapbox.navigation.base.utils.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.utils.DecodeUtils.stepGeometryToPoints
+import com.mapbox.navigation.utils.internal.logD
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * This function checks whether the [NavigationRoute] has unexpected closures, which could be a reason to re-route.
+ *
+ * `true` is returned whenever there's any [RouteLeg.closures] outside of a direct region around a coordinate from [RouteOptions.coordinatesList]
+ * that was allowed for snapping to closures with [RouteOptions.snappingIncludeClosuresList] or [RouteOptions.snappingIncludeStaticClosuresList].
+ * flags. Otherwise, `false` is returned.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+suspend fun NavigationRoute.hasUnexpectedClosures(): Boolean =
+    withContext(Dispatchers.IO) {
+        val LOG_CATEGORY = "HAS_ROUTE_UNEXPECTED_CLOSURES"
```

Comment:
```suggestion
        val LOG_CATEGORY = "NavigationRouteUtils"
```
and move to a `private const val`